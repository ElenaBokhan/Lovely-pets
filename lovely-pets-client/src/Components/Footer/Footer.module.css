.footer {
    height: 194px;
    display: flex;
    padding: 40px 0;
    justify-content: space-between;
    align-items: center;
}

.footer ul {
    list-style-type: none;
    width: 160px;
}

.footer ul li {
    line-height: 32px;
}

.footer ul li:hover {
    font-weight: bold;
}

.textColor {
    color: rgb(var(--text));
}

.link {
    position: relative;
}

.logo-block {
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: space-between;
}

.contacts {
    display: flex;
    flex-direction: column;
    height: 100%;
    width: 168px;
    justify-content: space-between;
}

.social {
    display: flex;
    justify-content: space-between;
    justify-self: flex-end;

}

.socialLogo *{
    fill: rgb(var(--text));
}

.bubble {
    background-color: #01a54e;
    border-radius: 50%;
    color: #fff;
    font-size: 12px;
    padding: 2px 7px;
    position: absolute;
    right: -20px;
    top: -10px;
    line-height: 18px;
}