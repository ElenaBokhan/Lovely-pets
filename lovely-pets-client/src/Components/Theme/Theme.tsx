import {createContext, useEffect, useState} from 'react';
import {THEME_STORAGE_KEY} from 'Const';

const supportedThemes = {
    light: 'light',
    dark: 'dark',
};

type Themes = keyof typeof supportedThemes;

export const ThemeContext = createContext<
    | {
          theme: Themes;
          setTheme: (theme: Themes) => void;
          supportedThemes: {[key: string]: string};
      }
    | undefined
>(undefined);

const getTheme = (): Themes => {
    let theme = localStorage.getItem(THEME_STORAGE_KEY);

    if (!theme) {
        localStorage.setItem(THEME_STORAGE_KEY, 'light');
        theme = 'light';
    }

    return theme as Themes;
};

const Theme = (props: {children: React.ReactNode}) => {
    const [theme, setTheme] = useState<Themes>(getTheme);

    useEffect(() => {
        localStorage.setItem(THEME_STORAGE_KEY, theme);
        document.documentElement.setAttribute('data-theme', theme);
    }, [theme]);

    return (
        <ThemeContext.Provider
            value={{
                theme,
                setTheme,
                supportedThemes,
            }}
        >
            {props.children}
        </ThemeContext.Provider>
    );
};

export default Theme;
