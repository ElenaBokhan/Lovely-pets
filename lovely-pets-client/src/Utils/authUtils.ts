import {ILoginResponse} from 'Api/authClient';

export const setLoginToStorage = (loginData: ILoginResponse) => {
    const {access_token, refresh_token, username, id} = loginData;

    localStorage.setItem('access_token', access_token);
    localStorage.setItem('refresh_token', refresh_token);
    localStorage.setItem('userName', username);
    localStorage.setItem('userId', id);
};

export const removeLoginToStorage = () => {
    localStorage.removeItem('access_token');
    localStorage.removeItem('refresh_token');
    localStorage.removeItem('userName');
    localStorage.removeItem('userId');
};
