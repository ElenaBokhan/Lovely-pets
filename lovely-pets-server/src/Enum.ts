export enum ESortFilter {
    POPULAR = 'POPULAR',
    NEW = 'NEW',
    CHEAPER = 'CHEAPER',
    EXPENSIVE = 'EXPENSIVE',
    RATING = 'RATING',
    DISCOUNT = 'DISCOUNT',
}
