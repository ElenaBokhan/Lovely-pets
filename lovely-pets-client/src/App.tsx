import './index.css';
import {Layout} from 'Components/Layout/Layout';
import {Catalog} from 'Pages/Catalog/Catalog';
import {Favourites} from 'Pages/Favourites/Favourites';
import {PageNotFound} from 'Pages/PageNotFound/PageNotFound';
import {PetCard} from 'Pages/PetCard/PetCard';
import {Profile} from 'Pages/Profile/Profile';
import {RouterProvider, createBrowserRouter, RouteObject} from 'react-router-dom';
import {ToastContainer} from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import {SignUp} from 'Pages/SignUp/SignUp';
import {SignIn} from 'Pages/SignIn/SignIn';
import {Cart} from 'Pages/Cart/Cart';
import {AddReview} from 'Pages/AddReview/AddReview';
import {AddProduct} from 'Pages/AddProduct/AddProduct';
import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
import {withTranslation} from 'react-i18next';
import {useEffect} from 'react';
import {setCartFromStorage, setCartToStorage} from 'Store/cartStore';
import 'Style/Themes.module.css';
import Theme from 'Components/Theme/Theme';

export const router: RouteObject[] = [
    {
        path: '/',
        element: <Layout />,
        errorElement: <div />, // TODO: добавить errorBoundary, пока заглушка
        children: [
            {
                path: '/',
                element: <Catalog />,
            },
            {
                path: '/favourites',
                element: <Favourites />,
            },
            {
                path: '/pets/:petId',
                element: <PetCard />,
            },
            {
                path: '/profile',
                element: <Profile />,
            },
            {
                path: '/addReview',
                element: <AddReview />,
            },
            {
                path: '/addPet',
                element: <AddProduct />,
            },
            {
                path: '/signup',
                element: <SignUp />,
            },
            {
                path: '/signin',
                element: <SignIn />,
            },
            {
                path: '/cart',
                element: <Cart />,
            },
            {
                path: '*',
                element: <PageNotFound />,
            },
        ],
    },
];

export const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const App = () => {
    useEffect(() => {
        window.addEventListener('load', setCartFromStorage);
        window.addEventListener('beforeunload', setCartToStorage);
        return () => {
            window.removeEventListener('load', setCartFromStorage);
            window.removeEventListener('beforeunload', setCartToStorage);
        };
    }, []);

    return (
        <QueryClientProvider client={queryClient}>
            <Theme>
                <RouterProvider router={createBrowserRouter(router)} />
                <ToastContainer autoClose={2000} position="top-right" theme="light" closeOnClick />
            </Theme>
        </QueryClientProvider>
    );
};

export default withTranslation()(App);
