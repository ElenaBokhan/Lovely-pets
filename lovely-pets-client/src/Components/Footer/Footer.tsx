import {ReactComponent as InstagramIcon} from 'assets/logo-instagram.svg';
import {ReactComponent as VkIcon} from 'assets/logo-vk.svg';
import {ReactComponent as TgIcon} from 'assets/telegram.svg';
import {ReactComponent as ViberIcon} from 'assets/viber.svg';
import {ReactComponent as WhatsappIcon} from 'assets/whatsapp.svg';
import FooterContainer, {EContainerType} from 'Components/Common/Container/Container';
import {IconButton} from 'Components/Common/IconButton/IconButton';
import {EFontWeight, ETextType, Text} from 'Components/Common/Text/Text';
import styles from 'Components/Footer/Footer.module.css';
import {ETestId} from 'Enum';
import {useTranslation} from 'react-i18next';
import {Link} from 'react-router-dom';
import {useState} from 'react';
import {cartStore} from 'Store/cartStore';
import {useStore} from '@tanstack/react-store';
import {Logo} from '../Common/Logo/Logo';

enum EMenuItem {
    CATALOG = 'CATALOG',
    FAVOURITES = 'FAVOURITES',
    CART = 'CART',
    USER_PROFILE = 'USER_PROFILE',
}

export const Footer = () => {
    const [selectedMenuItem, setSelectedMenuItem] = useState<keyof typeof EMenuItem>(EMenuItem.CATALOG);
    const {t} = useTranslation();

    const cartCount = useStore(cartStore, (state) => Object.keys(state).length);

    const handleChangeMenuItem = (event: React.MouseEvent<HTMLAnchorElement>) => {
        setSelectedMenuItem(event.currentTarget.dataset.itemId as keyof typeof EMenuItem);
    };

    const renderFooterLogo = () => (
        <div className={styles['logo-block']}>
            <Logo />
            <Text type={ETextType.S2} value={'© «Питомник домашних котиков lovelyPets.ru»'} />
        </div>
    );
    const renderMenu = () => {
        const menuConfig = Object.keys(EMenuItem).map((item) => {
            const menuItem = item.toLowerCase();

            return {
                key: item,
                label: t(`footer.menu.${menuItem}`),
                path: `/${item === EMenuItem.CATALOG ? '' : menuItem}`,
            };
        });

        const fakeMenu = [
            t('footer.fake_menu.payment_and_delivery'),
            t('footer.fake_menu.often_asked'),
            t('footer.fake_menu.feedback'),
            t('footer.fake_menu.contacts'),
        ];

        return (
            <>
                <ul data-test-id={ETestId.FOOTER_MENU}>
                    {menuConfig.map(({key, label, path}) => {
                        const showBubble = !!cartCount && key === EMenuItem.CART;

                        return (
                            <li key={label}>
                                <Link
                                    className={styles.link}
                                    to={path}
                                    data-item-id={key}
                                    onClick={handleChangeMenuItem}
                                >
                                    {showBubble && <div className={styles.bubble}>{cartCount}</div>}
                                    <Text
                                        className={styles.textColor}
                                        type={ETextType.P2}
                                        value={label}
                                        weight={selectedMenuItem === key ? EFontWeight.GENERAL : EFontWeight.SECONDARY}
                                    />
                                </Link>
                            </li>
                        );
                    })}
                </ul>

                <ul data-testid={ETestId.FOOTER_MENU}>
                    {fakeMenu.map((menuItem) => {
                        return (
                            <li key={menuItem}>
                                <Text className={styles.textColor} type={ETextType.P2} value={menuItem} />
                            </li>
                        );
                    })}
                </ul>
            </>
        );
    };

    const renderContactsGroup = () => (
        <div data-test-id={ETestId.FOOTER_CONTACTS} className={styles.contacts}>
            <Text value={t('footer.in_contact')} weight={EFontWeight.GENERAL} />
            <Text value={'8 (999) 00-00-00'} />
            <Text type={ETextType.S1} value={'lovelyPets@gmail.com'} />
            <div className={styles.social}>
                <IconButton>
                    <TgIcon className={styles.socialLogo} />
                </IconButton>
                <IconButton>
                    <WhatsappIcon className={styles.socialLogo} />
                </IconButton>
                <IconButton>
                    <ViberIcon className={styles.socialLogo} />
                </IconButton>
                <IconButton>
                    <InstagramIcon className={styles.socialLogo} />
                </IconButton>
                <IconButton>
                    <VkIcon className={styles.socialLogo} />
                </IconButton>
            </div>
        </div>
    );

    return (
        <FooterContainer className={styles.footer} type={EContainerType.FOOTER}>
            {renderFooterLogo()}
            {renderMenu()}
            {renderContactsGroup()}
        </FooterContainer>
    );
};
