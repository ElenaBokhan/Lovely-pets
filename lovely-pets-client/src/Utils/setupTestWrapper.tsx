import {queryClient, router} from 'App';
import {render} from '@testing-library/react';
import {createMemoryRouter, RouterProvider} from 'react-router-dom';
import {ToastContainer} from 'react-toastify';
import {QueryClientProvider} from '@tanstack/react-query';
import Theme from 'Components/Theme/Theme';

interface WrapperForTestsParams {
    initialRouterEntries: string[];
}

export const setupTestsWrapper = ({initialRouterEntries}: WrapperForTestsParams) => {
    const memoryRouter = createMemoryRouter(router, {
        initialEntries: initialRouterEntries,
    });
    render(
        <QueryClientProvider client={queryClient}>
            <Theme>
                <RouterProvider router={memoryRouter} />
                <ToastContainer autoClose={2000} position="top-right" theme="light" closeOnClick />
            </Theme>
        </QueryClientProvider>
    );

    return memoryRouter;
};
