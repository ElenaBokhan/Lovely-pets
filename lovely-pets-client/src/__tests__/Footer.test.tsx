import '@testing-library/jest-dom';
import {setupTestsWrapper} from 'Utils/setupTestWrapper';
import {getByTestId, render, screen} from '@testing-library/react';
import {ETestId} from 'Enum';
import {Footer} from 'Components/Footer/Footer';
import {queryClient} from 'App';
import {QueryClientProvider} from '@tanstack/react-query';
import {BrowserRouter} from 'react-router-dom';
import Theme from 'Components/Theme/Theme';

describe('Тесты для компонента Footer', () => {
    beforeEach(() => {
        setupTestsWrapper({initialRouterEntries: ['/']});
    });

    describe('Проверяем доступность элементов', () => {
        test('Проверяем доступность логотипа', () => {
            const footer = screen.getByTestId(ETestId.FOOTER);
            const footerLogo = getByTestId(footer, ETestId.MAIN_LOGO);
            expect(footerLogo).toBeInTheDocument();
        });

        test('Проверяем доступность меню', () => {
            expect(screen.getByTestId(ETestId.FOOTER_MENU)).toBeInTheDocument();
        });

        test('Проверяем доступность блока с контактами', () => {
            expect(screen.getByTestId(ETestId.FOOTER_CONTACTS)).toBeInTheDocument();
        });
    });

    describe('Снапшот футера', () => {
        test('Snapshot', () => {
            const {asFragment} = render(
                <QueryClientProvider client={queryClient}>
                    <Theme>
                        <BrowserRouter basename="/">
                            <Footer />
                        </BrowserRouter>
                    </Theme>
                </QueryClientProvider>
            );

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
