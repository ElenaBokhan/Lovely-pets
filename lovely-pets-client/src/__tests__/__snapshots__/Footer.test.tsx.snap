// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты для компонента Footer Снапшот футера Snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footerContainer"
    data-test-id="footer"
  >
    <div
      class="container footer"
    >
      <div
        class="logo-block"
      >
        <div
          class="logoWrapper"
        >
          <div
            alt="logoMain"
            class="logo"
            data-test-id="mainLogo"
            height="56"
          />
          <span
            class="undefined bold head1"
          >
            LovelyPets
          </span>
        </div>
        <span
          class="undefined undefined subscribe2"
        >
          © «Питомник домашних котиков lovelyPets.ru»
        </span>
      </div>
      <ul
        data-test-id="footerMenu"
      >
        <li>
          <a
            class="link"
            data-item-id="CATALOG"
            href="/"
          >
            <span
              class="textColor undefined bold paragraph2"
            >
              Каталог
            </span>
          </a>
        </li>
        <li>
          <a
            class="link"
            data-item-id="FAVOURITES"
            href="/favourites"
          >
            <span
              class="textColor undefined normal paragraph2"
            >
              Избранные
            </span>
          </a>
        </li>
        <li>
          <a
            class="link"
            data-item-id="CART"
            href="/cart"
          >
            <span
              class="textColor undefined normal paragraph2"
            >
              Корзина
            </span>
          </a>
        </li>
        <li>
          <a
            class="link"
            data-item-id="USER_PROFILE"
            href="/user_profile"
          >
            <span
              class="textColor undefined normal paragraph2"
            >
              Профиль
            </span>
          </a>
        </li>
      </ul>
      <ul
        data-testid="footerMenu"
      >
        <li>
          <span
            class="textColor undefined undefined paragraph2"
          >
            Оплата и доставка
          </span>
        </li>
        <li>
          <span
            class="textColor undefined undefined paragraph2"
          >
            Часто спрашивают
          </span>
        </li>
        <li>
          <span
            class="textColor undefined undefined paragraph2"
          >
            Обратная связь
          </span>
        </li>
        <li>
          <span
            class="textColor undefined undefined paragraph2"
          >
            Контакты
          </span>
        </li>
      </ul>
      <div
        class="contacts"
        data-test-id="footerContacts"
      >
        <span
          class="undefined bold paragraph1"
        >
          Мы на связи
        </span>
        <span
          class="undefined undefined paragraph1"
        >
          8 (999) 00-00-00
        </span>
        <span
          class="undefined undefined subscribe1"
        >
          lovelyPets@gmail.com
        </span>
        <div
          class="social"
        >
          <button
            class="button"
            type="button"
          >
            <div
              class="socialLogo"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <div
              class="socialLogo"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <div
              class="socialLogo"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <div
              class="socialLogo"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <div
              class="socialLogo"
            />
          </button>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
